projects/inkscape/build.sh

```diff
@@ -0,0 +1,64 @@
+cd ../atkmm* && ./configure --prefix=/usr --enable-static  && make -j${nproc} && make install
+cd ../pangomm* && ./configure --prefix=/usr --enable-static  && make -j${nproc} && make install
+cd ../gtkmm* && ./configure --prefix=/usr --enable-static  && make -j${nproc} && make install
+cd ../gdl-3* && ./configure --prefix=/usr --enable-static  && make -j${nproc} && make install
+cd ../dbus* && ./configure --prefix=/usr --enable-static && make -j${nproc} && make install
+cd ../inkscape #at long last
+
+###   build inkscape ###
+
+mkdir build
+cd build
+CFLAGS=${OLDCFLAGS}
+CXXFLAGS=${OLDCXXFLAGS}
+SANITIZER=${OLDSANITIZER}
+COVERAGE_FLAGS=${OLDCOVERAGE_FLAGS}
+cmake .. -DWITH_FUZZ=ON -DLIB_FUZZING_ENGINE=ON -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX:PATH=$PWD/install_dir/ -DBUILD_SHARED_LIBS=OFF -DCMAKE_POSITION_INDEPENDENT_CODE=OFF \
+    -DENABLE_LCMS=OFF -DENABLE_POPPLER=OFF -DENABLE_POPPLER_CAIRO=OFF -DWITH_IMAGE_MAGICK=OFF -DWITH_LIBCDR=OFF -DWITH_LIBVISIO=OFF -DWITH_LIBWPG=OFF -DWITH_LPETOOL=OFF -DWITH_NLS=OFF -DWITH_OPENMP=OFF -DWITH_YAML=OFF
+make -j$(nproc) inkscape_base
+cd testfiles/
+
+###   compile fuzzer   ###
+
+clang++ -I/src/inkscape/build/testfiles -I/src/inkscape/testfiles -I/src/inkscape -I/src/inkscape/src -I/src/inkscape/build/include \
+    -isystem /usr/include/pango-1.0 -isystem /usr/include/cairo -isystem /usr/include/pixman-1 -isystem /usr/include/libpng12 -isystem /usr/include/harfbuzz -isystem /usr/include/freetype2 -isystem /usr/include/libsoup-2.4 -isystem /usr/include/libxml2 -isystem /usr/include/glib-2.0 -isystem /usr/lib/x86_64-linux-gnu/glib-2.0/include -isystem /usr/include/gc -isystem /usr/include/dbus-1.0 -isystem /usr/lib/x86_64-linux-gnu/dbus-1.0/include -isystem /usr/include/gio-unix-2.0 -isystem /usr/include/gtkmm-3.0 -isystem /usr/lib/gtkmm-3.0/include -isystem /usr/include/atkmm-1.6 -isystem /usr/include/gtk-3.0/unix-print -isystem /usr/include/gdkmm-3.0 -isystem /usr/lib/gdkmm-3.0/include -isystem /usr/include/giomm-2.4 -isystem /usr/lib/giomm-2.4/include -isystem /usr/include/pangomm-1.4 -isystem /usr/lib/pangomm-1.4/include -isystem /usr/include/glibmm-2.4 -isystem /usr/lib/glibmm-2.4/include -isystem /usr/include/cairomm-1.0 -isystem /usr/lib/cairomm-1.0/include -isystem /usr/include/sigc++-2.0 -isystem /usr/lib/sigc++-2.0/include -isystem /usr/include/libgdl-3.0 -isystem /usr/include/gtk-3.0 -isystem /usr/include/at-spi2-atk/2.0 -isystem /usr/include/at-spi-2.0 -isystem /usr/include/atk-1.0 -isystem /usr/include/gdk-pixbuf-2.0 -isystem /usr/src/gmock/gtest/include -isystem /usr/src/gmock/include -isystem /src/inkscape/gtest/gtest/include  \
+    ${CXXFLAGS} \
+    -lpthread -std=c++11 -stdlib=libc++ -g -UWITH_LPETOOL -ULPE_ENABLE_TEST_EFFECTS -o CMakeFiles/fuzz.dir/fuzzer.cpp.o -c /src/inkscape/testfiles/fuzzer.cpp
+
+###   link fuzzer   ###
+
+clang++ ${CXXFLAGS} \
```

Comment:
use $CXX instead of clang++ (same thing currently, but $CXX is what we document as the interface)

Reply:
Also, I don't see $LIB_FUZZING_ENGINE here... Do you actually link with the fuzzing engine? 
If this links, someone else is providing `main()` (which is wrong)

Reply:
It's in the "/usr/lib/*.a" (that contains everything I manually builds)
I used it to make sure that it would not use /usr/x86_64/ version for one of them (iirc, my version was originally shadowed by it for the linker) )

Reply:
Hm... But I don't see you adding "/usr/lib/*.a". And it looks fragile, it's better to $LIB_FUZZING_ENGINE explicitly. 
ALso, some of your `-l` options sound strange. 
Things like `-lc++ -lc -lr -lgcc_s -lgcc -lselinux -lpthread -lsystemd`. I am always suspicious when I see these added explicitly. 

(It's very sad we have to do it this way -- I would have preferred to build smaller parts of the code with fewer deps -- but I already complained :()

Reply:
It's on the link command, at the very end of line 59. , but ok to add -lFuzzingEngine
Basically, I took the lines generated by CMake, the first one that generate the .o and works fine, and the "link" line that I tweaked for days to add all the deps until it stopped complaining. I did not understand why it asked me to add those (in almost all examples I saw elsewhere, they are not needed) but their absence provoked an error…

I would also have preferred a simpler way without any of the UI code but there are not really any way to build Inkscape without it currently :/ (as soon as there is, I'll update/simplify the code here^^)